frontend/src/app/libs/container/components/header/header.component.ts

```diff
@@ -0,0 +1,89 @@
+import { Component, ElementRef, OnDestroy, OnInit, QueryList, ViewChildren } from '@angular/core';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+import { EntityWrapper } from 'src/app/libs/auth/models/entity-wraper';
+import { User } from 'src/app/libs/auth/models/user';
+import { ButtonType } from 'src/app/libs/common-components/components/button/models/button-type.enum';
+import { ComponentTheme } from 'src/app/libs/common-components/shared/component-theme.enum';
+import { Localization } from '../../models/localization';
+import { Claim } from '../../models/claim';
+import { ContainerFacadeService } from '../../services/container.facade';
+import { LocalizationService } from '../../services/localization.service';
+import { ButtonComponent } from 'src/app/libs/common-components/components/button/button.component';
+
+@Component({
+  selector: 'app-header',
+  templateUrl: './header.component.html',
+  styleUrls: ['./header.component.scss']
+})
+export class HeaderComponent implements OnInit, OnDestroy {
+
+	public outlinedButtonType: ButtonType = ButtonType.Outlined;
+	public flatButtonType: ButtonType = ButtonType.Flat;
+
+	public darkTheme: ComponentTheme = ComponentTheme.Dark;
+
+	public user: User;
+
+	public adminLinksActivation: Map<string, boolean> = new Map([
+		['events', true],
+		['administration', false]
+	]);
+
+	public unsubscribe$: Subject<boolean> = new Subject<boolean>();
+
+	public get userName(): string {
+		if (this.user?.firstName.replace(/\s/g, '').length > 0 || this.user?.secondName.replace(/\s/g, '').length > 0) {
+			return `${this.user?.firstName} ${this.user?.secondName}`;
+		} else {
+			return this.user.email;
+		}
+	}
+
+	public get localization(): Localization {
+		return this.localizationService.currentLocalization;
+	}
+
+	public get isUnauthorizedUser(): boolean {
+		return this.user === undefined;
+	}
+
+	public get isAdminUser(): boolean {
+		return this.user ? this.user.claims?.includes(Claim.Admin) : false;
```

Comment:
please make Claims related code parts similar with https://github.com/GandalfClub/gandalf-app/pull/58

Reply:
also seems like we making components to fat, please move all user related logic to user-service which will calculate user name, is admin and etc

Reply:
also make sure userName composer method is utility and can me reused somewhere outside